model/src/cfd.rs

```diff
@@ -143,6 +146,27 @@ impl MakerOffers {
         None
     }
 
+    /// Takes the order if available
+    ///
+    /// Resets the order that was taken to None.
```

Comment:
instead of resetting it to `None`, this would be more useful if it just called `replicate()` on an Order to create a new one! 

this `None` is not used anywhere, and if we replicated it here, we wouldn't need to replicate the whole lot (I did replication of both just to speed up) . 

I'm happy to do it as a follow-up, it's non-blocking change - just let me know if you disagree :)

Reply:
This function is called by the taker when taking an order. The taker cannot replicate orders, because we currently take orders by id, and if the taker replicates the maker would not know anything about the ids being created. 
The taker knows that that a take request was sent and hence should set to `None` - but cannot replicate. Eventually we should re-visit the old discussion if it's a good idea to take by order-id...

Reply:
thanks for the explanation!